### PR TITLE
fix(angular): add cypress workaround when generating mfes

### LINKED ISSUE #8081: Angular 13 and MFE breaks with Cypress
## Current Behavior

The `styles` defined in `angular.json` cannot be injected `default value = inject:true`.
Causes the test to fail completely.

## Steps to Reproduce
1. Create a **fresh** nx workspace
2. add an angular app with mfe preconfigured
3. run e2e tests.

The nx custom webpack config breaks the tests. Remove the custom webpack config and the e2e tests work again.

### Failure Logs

```
       "before each" hook for "should display welcome message":
     SyntaxError: The following error originated from your application code, not from Cypress.

  > Cannot use 'import.meta' outside a module

When Cypress detects uncaught errors originating from your application it will automatically fail the current test.

This behavior is configurable, and you can choose to turn this off by listening to the `uncaught:exception` event.

https://on.cypress.io/uncaught-exception-from-application
```

### Environment

```
>  NX  Report complete - copy this into the issue template

  Node : 14.16.0
  OS   : darwin x64
  npm  : 6.14.11
  
  nx : 13.2.4
  @nrwl/angular : 13.2.4
  @nrwl/cli : 13.2.4
  @nrwl/cypress : 13.2.4
  @nrwl/devkit : 13.2.4
  @nrwl/eslint-plugin-nx : 13.2.4
  @nrwl/express : undefined
  @nrwl/jest : 13.2.4
  @nrwl/linter : 13.2.4
  @nrwl/nest : undefined
  @nrwl/next : undefined
  @nrwl/node : undefined
  @nrwl/nx-cloud : undefined
  @nrwl/react : undefined
  @nrwl/react-native : undefined
  @nrwl/schematics : undefined
  @nrwl/tao : 13.2.4
  @nrwl/web : undefined
  @nrwl/workspace : 13.2.4
  @nrwl/storybook : 13.2.4
  @nrwl/gatsby : undefined
  typescript : 4.4.4
  rxjs : 7.4.0
  ---------------------------------------
  Community plugins:
         @angular-architects/module-federation: 12.5.3
         @angular/animations: 13.0.3
         @angular/common: 13.0.3
         @angular/compiler: 13.0.3
         @angular/core: 13.0.3
         @angular/forms: 13.0.3
         @angular/platform-browser: 13.0.3
         @angular/platform-browser-dynamic: 13.0.3
         @angular/router: 13.0.3
         @angular-devkit/build-angular: 13.0.4
         @angular/cli: 13.0.4
         @angular/compiler-cli: 13.0.3
         @angular/language-service: 13.0.3
```

### PR DESCRIPTION
**Pull request by Coly010**
## Current Behavior

Cypress falls over when testing MFEs because an error is output to the console relating to how the styles.js is attached to index.html in dev mode.

## Expected Behavior

Temporary workaround is to tell cypress to ignore the error.

## Related Issue(s)

Fixes #8081

### CODE REVIEW

File: packages/angular/src/generators/setup-mfe/lib/add-cypress-workaround.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    e2eProject.sourceRoot,
+    'support/commands.ts'
+  );
+
+  if (!tree.exists(pathToCommandsFile)) {
+    tree.write(pathToCommandsFile, '');
+  }
+
+  const commandsContent = tree.read(pathToCommandsFile, 'utf-8');
+  tree.write(pathToCommandsFile, `${commandsContent}\n${commandToAdd}`);
```
Comment (leosvelperez) [inline packages/angular/src/generators/setup-mfe/lib/add-cypress-workaround.ts:48]: We could simplify this and avoid writing the file first with an empty string to then read that empty string:

```suggestion
  const commandsContent = tree.exists(pathToCommandsFile)
    ? tree.read(pathToCommandsFile, 'utf-8')
    : '';
  tree.write(pathToCommandsFile, `${commandsContent}\n${commandToAdd}`);
```

File: packages/angular/src/generators/setup-mfe/schema.json [hunk 2]
```diff
@@ -45,6 +45,10 @@
       "type": "boolean",
       "default": false,
       "description": "Do not add dependencies to `package.json`."
+    },
+    "e2eProjectName": {
+      "type": "string",
+      "description": "The project name of the associated e2e project for the application."
```
Comment (leosvelperez) [inline packages/angular/src/generators/setup-mfe/schema.json:51]: Maybe we could also say that's only needed for Cypress e2e projects with a name that doesn't follow the `<app>-e2e` naming convention.

**Review by leosvelperez — CHANGES_REQUESTED** [review]

File: packages/angular/src/generators/setup-mfe/lib/add-cypress-workaround.ts (hunk 1 above)
Comment (Coly010) [reply packages/angular/src/generators/setup-mfe/lib/add-cypress-workaround.ts:48]: i like it!

File: packages/angular/src/generators/setup-mfe/schema.json (hunk 2 above)
Comment (Coly010) [reply packages/angular/src/generators/setup-mfe/schema.json:51]: yeah good idea

File: packages/angular/src/generators/setup-mfe/lib/add-cypress-workaround.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  const commandToAdd = `Cypress.on('uncaught:exception', err => {
+    if (err.message.includes(\`Cannot use 'import.meta' outside a module\`)) {
+      return false;
+    }
+    return true;
+  });`;
+
+  const pathToCommandsFile = joinPathFragments(
+    e2eProject.sourceRoot,
+    'support/commands.ts'
```
[the remark this answers is not in the document]
Comment (Coly010) [reply packages/angular/src/generators/setup-mfe/lib/add-cypress-workaround.ts:40]: ah fair, ok!

File: packages/angular/migrations.json [hunk 4]
```diff
@@ -144,6 +144,12 @@
       "version": "13.5.0-beta.0",
       "description": "A lot of changes to how MFEs operate were discovered and merged without appropriate migrations. This should cover migrating existing MFEs to the latest, using ESM and optmized production bundles.",
       "factory": "./src/migrations/update-13-5-0/update-mfe-configs"
+    },
+    "add-cypress-mfe-workaround": {
+      "cli": "nx",
+      "version": "13.9.0-beta.0",
```
Comment (Coly010) [inline packages/angular/migrations.json:150]: Not sure what version to target. I'd like this out sooner if possible I guess

File: packages/angular/src/generators/setup-mfe/schema.json [hunk 5]
```diff
@@ -45,6 +45,10 @@
       "type": "boolean",
       "default": false,
       "description": "Do not add dependencies to `package.json`."
+    },
+    "e2eProjectName": {
+      "type": "string",
+      "description": "The project name of the associated e2e project for the application. This is only required for Cypress e2e projects that do no follow the naming convention <appName>-e2e"
```
Comment (leosvelperez) [inline packages/angular/src/generators/setup-mfe/schema.json:51]:
```suggestion
      "description": "The project name of the associated e2e project for the application. This is only required for Cypress e2e projects that do not follow the naming convention `<appName>-e2e`."
```

**Review by leosvelperez — COMMENTED** [review]
Left just one last minor comment. It should be good to go after that.

You could ask Jason whether a patch version would be released with it or if it would be in the next minor version.

**Review by leosvelperez — APPROVED** [review]
LGTM! 🎉

**Pull request merged**